src/components/templates/discount-general/index.tsx

```diff
@@ -0,0 +1,209 @@
+  onStatusChange?: () => void
+}
+
+const DiscountGeneral: React.FC<DiscountGeneralProps> = ({
+  subtitle,
+  isEdit,
+  regionOptions,
+  selectedRegions,
+  setSelectedRegions,
+  discountType,
+  setDiscountType,
+  isDisabled,
+  isFreeShipping,
+  isDynamic,
+  setIsDynamic,
+  regionIsDisabled = false,
+  onDelete,
+  onDuplicate,
+  onStatusChange,
+}) => {
+  const editActions: ActionType[] = [
+    {
+      label: "Dublicate",
+      onClick: () => {
+        if (onDuplicate) {
+          onDuplicate()
+        }
+      },
+      icon: <DuplicateIcon size={20} />,
+    },
+    {
+      label: "Delete",
+      onClick: () => {
+        if (onDelete) onDelete()
+      },
+      icon: <TrashIcon size={20} />,
+      variant: "danger",
+    },
+  ]
+
+  return (
+    <BodyCard
+      title="General"
+      subtitle={subtitle}
+      className="h-auto"
+      actionables={isEdit ? editActions : undefined}
+      status={
+        isEdit ? (
+          <StatusSelector
+            isDraft={isDisabled}
+            activeState="Active"
+            draftState="Draft"
+            onChange={() => {
+              if (onStatusChange) onStatusChange()
+            }}
+          />
+        ) : undefined
+      }
+    >
+      <div className="mt-large">
+        <ConnectForm>
+          {({ register }) => (
+            <div>
+              <h3 className="inter-base-semibold mb-2xsmall">
+                General information
+              </h3>
+              <p className="inter-small-regular text-grey-50">
+                The code your customers will enter during checkout. Uppercase
+                letters and numbers only.
+              </p>
+              <div className="flex gap-x-xlarge mt-base w-full">
+                <div className="flex flex-col w-1/2 gap-y-base">
+                  <InputField
+                    label="Code"
+                    placeholder="SUMMERSALE10"
+                    required
+                    name="code"
+                    ref={register({ required: true })}
+                  />
+                  <Select
+                    label="Choose valid regions"
+                    options={regionOptions}
+                    isMultiSelect
+                    enableSearch
+                    value={selectedRegions}
+                    onChange={setSelectedRegions}
+                    required
+                    hasSelectAll
+                    className={clsx({
+                      ["opacity-50 pointer-events-none select-none"]: regionIsDisabled,
+                    })}
+                    disabled={regionIsDisabled}
+                  />
+                  <InputField
+                    label="Amount"
+                    required
+                    type="number"
+                    placeholder="10"
+                    prefix={discountType === "percentage" ? "%" : undefined}
+                    name="rule.value"
+                    ref={register({ required: true })}
```

Comment:
Should not be required if discount is free shipping, since I think this prevents us from saving the discount. At least that's the behaviour, I am encountering atm

Reply:
@olivermrbl  Actually think this is due to an error in the validator used in the API that requires that amount is a number always 😅 

```ts
@IsNumber()
  value: number
```

I fixed it for now by setting the value to 0 if free shipping is selected :smile